jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -2166,7 +2166,7 @@ public String toGenericString() {
  */
 public Package getPackage() {
 	String packageName = getPackageName();
-	if (packageName == null) {
+	if ((null == packageName) || isArray() || isPrimitive()) {
```

Comment:
`getPackageName` will have called `isPrimitive` and may have already called `isArray`: perhaps it would be better to reorder the calls here so `getPackageName` is called after we've ruled out primitive and array types?

Could you also correct the javadoc for `getPackageName`?

Reply:
Good plan.  Done.